napari/layers/_image_layer/model.py

```diff
@@ -247,16 +259,28 @@ def interpolation_index(self, interpolation_index):
     def colormap(self):
         """string or ColorMap: Colormap to use for luminance images.
         """
-        return self._node.cmap
+        return self.colormap_name, self._node.cmap
 
     @colormap.setter
     def colormap(self, colormap):
-        try:
-            cmap = self._colormaps[colormap]
-        except KeyError:
-            pass
-
-        self._node.cmap = cmap
+        name = '[unnamed colormap]'
+        if isinstance(colormap, str):
+            name = colormap
+        elif isinstance(colormap, tuple):
+            name, cmap = colormap
+            self._colormaps[name] = cmap
```

Comment:
What if someone wants to add a colormap with the same name as an existing one? Do we want to increment its name or do we want to replace it entirely?

Reply:
I'd say probably fine to replace it entirely

Reply:
Yes, I think replace is fine. Until we get our first user complaint at least. =P